js/main.js

```diff
@@ -1,8 +1,11 @@
+var $h1 = document.querySelector('h1');
+var $form = document.querySelector('#form');
+var $table = document.querySelector('table');
 var $tableLast = document.querySelectorAll('.last');
 var $tableTitle = document.querySelector('.title');
-var $tableData = document.querySelectorAll('td')
-
-var $form = document.querySelector('#form');
+var $tableData = document.querySelectorAll('td');
+var $p = document.querySelector('p');
+var $a = document.querySelectorAll('a');
```

Comment:
Referencing DOM elements by tag name (particularly common ones like `<p>` and `<a>`) is fragile. Adding or removing elements to/from the page can easily break the logic. It'd be better to utilize ids (for unique elements) or class names (for lists of elements).